File: examples/python/evaluators/basic/default_preset.py
from typing import Dict, Union, Any


def evaluate(
    app_params: Dict[str, str],
    inputs: Dict[str, str],
    output: Union[str, Dict[str, Any]],  # output of the llm app
    correct_answer: str,  # contains the testset row
) -> float:
    if output in correct_answer:
        return 1.0
    else:
        return 0.0

File: examples/javascript/evaluators/basic/default_preset.js
/**
 * Character Count Match Test (JavaScript)
 * ======================================
 *
 * Simple evaluator that compares character counts for output vs correct answer.
 * This mirrors the Python exact_match example without NumPy.
 */

function evaluate(appParams, inputs, output, correctAnswer) {
  void appParams
  void inputs

  try {
    const outputStr =
      typeof output === "string" ? output : JSON.stringify(output)
    const answerStr = String(correctAnswer)

    return outputStr.length === answerStr.length ? 1.0 : 0.0
  } catch {
    return 0.0
  }
}
